pkg/cli/cli.go

```diff
@@ -910,3 +925,52 @@ func buildServiceRootEndpoint(serviceURL, zone string) string {
 	const version = "v1"
 	return client.BuildRootEndpoint(serviceURL, version, zone)
 }
+
+type stateOutput struct {
+	// State description.
+	Message string
+	// True if state is completed.
+	Done bool
+}
+
+// Prints out state changes.
+//
+// Only use this printer to print user friendly state changes, this is not a logger.
+type statePrinter struct {
+	// Writer to print the messages to. Visual features won't be displayed if not linked to an interactive terminal.
+	Out io.Writer
+	// If true, visual features won't be displayed.
+	Verbose bool
+
+	// If true, visual features like colors and animations won't be displayed.
+	visualsOff bool
+}
+
+func newStatePrinter(out io.Writer, verbose bool) *statePrinter {
+	visualsOff := false
+	if f, ok := out.(*os.File); !ok {
+		visualsOff = true
+	} else if !term.IsTerminal(int(f.Fd())) {
+		visualsOff = true
+	}
+	if verbose {
+		visualsOff = true
+	}
+	return &statePrinter{Out: out, Verbose: verbose, visualsOff: visualsOff}
```

Comment:
`Verbose` seems to be used only to turn visuals off. If that the case there is no need to store it in the struct.

Reply:
agree, removed.